app/models/step.rb

```diff
@@ -2,7 +2,7 @@ class Step < ApplicationRecord
   belongs_to :step_by_step_page
   validates_presence_of :step_by_step_page
   validates :title, :logic, presence: true
-  has_many :link_reports
+  has_many :link_reports, :dependent => :destroy
```

Comment:
Same comment as on the other PR - should we be explicit about this in a model test?

Reply:
I cherry-picked in the commit from the other PR to get the tests to work. I'll unpick it when the other PR is done/merged.